src/alspgrad.jl

```diff
@@ -47,7 +47,7 @@ immutable ALSGradUpdH_State{T}
     WtX::Matrix{T}    # W'X  (pre-computed)
     WtWD::Matrix{T}   # W'W * D
 
-    function ALSGradUpdH_State(X, W, H)
+    function ALSGradUpdH_State{T}(X, W, H) where T <: Real
```

Comment:
It seems this was still in. How should one handle v0.5/v0.6 incompatible syntax??